ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/akkastreams/dispatcher/Dispatcher.scala

```diff
@@ -43,10 +43,15 @@ trait Dispatcher[Index] {
     */
   def shutdown(): Future[Unit]
 
-  /** Triggers shutdown of the Dispatcher by eagerly failing
-    * all outstanding stream subscriptions with the given throwable.
+  /** Triggers shutdown of the Dispatcher by eagerly failing all outstanding stream subscriptions with a throwable.
+    *
+    * @param newThrowable Create a new throwable.
+    *                     It is important to create a new throwable for each failed subscription
+    *                     since the throwable closing the streams or parts of it can be mutable.
+    *                     (e.g. the [[io.grpc.Metadata]] provided as part of [[io.grpc.StatusRuntimeException]]
+    *                     is mutated in the gRPC layer)
     */
-  def cancel(throwable: Throwable): Future[Unit]
+  def cancel(newThrowable: () => Throwable): Future[Unit]
```

Comment:
(just an idea)
I have looked at the caller side of this method and found only one particular case with one very specific Throwable being passed here... I am wondering - what if we just simplify this thing and instead of this rather complicated argument with closure just provide void and cancel will always throw `LedgerApiErrors.ServiceNotRunning.Reject` ?

Reply:
yeah, for `Dispatcher` I could do that since it's indeed only one exception envisioned to be used. But not sure for `SignalDispatcher` 🤔

Reply:
But is it called from the other callers apart from `DispatcherImpl`? I think if we change `Dispatcher` then we would do the same to `SignalDispatcher.fail`

Reply:
Not yet, but the `SignalDispatcher` has other uses than `DispatcherImpl` (see `ApiTimeService`) and it's generally-purpose built.

Reply:
Thank you for the clarification, this is a good point.